app/uk/gov/hmrc/gform/scheduler/QueueAlgebra.scala

```diff
@@ -26,14 +27,17 @@ trait QueueAlgebra[Q] {
   val pollLimit: Int
   val maxFailureCount: Int
   val retryIntervalMillis: Long
+  val logger: Logger
 
   implicit val executionContext: ExecutionContext
 
   def sendWorkItem(workItem: WorkItem[Q]): Future[Unit]
 
   def processThenMarkAsComplete(acc: Seq[WorkItem[Q]], workItem: WorkItem[Q]): Future[Seq[WorkItem[Q]]] =
     sendWorkItem(workItem).map(_ => repo.completeAndDelete(workItem.id)).map(_ => acc :+ workItem).recoverWith {
-      case _ => repo.failed(workItem, maxFailureCount).map(_ => acc)
+      case e =>
+        logger.error(s"Failed to send the work-item : $e")
```

Comment:
I wonder if this shouldn't be rather this:

```suggestion
        logger.error(s"Failed to send the work-item", e)
```